backend/shork/src/test/kotlin/compiler/TestParser.kt

```diff
@@ -614,5 +622,353 @@ internal class TestParser {
+                    Triple(TokenType.JMP, Jmp::class, Modifier.B),
+                    Triple(TokenType.JMZ, Jmz::class, Modifier.B),
+                    Triple(TokenType.JMN, Jmn::class, Modifier.B),
+                    Triple(TokenType.DJN, Djn::class, Modifier.B),
+                    Triple(TokenType.SPL, Spl::class, Modifier.B),
+                )
+
+            for (instruction in instructionWithModifier) {
+                for (firstAddressMode in addressModes) {
+                    for (secondAddressMode in addressModes) {
+                        val tokens =
+                            listOf(
+                                Token(instruction.first, instruction.first.toString(), "", 1, 0, 0),
+                                Token(firstAddressMode.third, firstAddressMode.second, "", 1, 0, 0),
+                                Token(
+                                    TokenType.NUMBER,
+                                    firstAddress.toString(),
+                                    firstAddress,
+                                    1,
+                                    0,
+                                    0,
+                                ),
+                                Token(TokenType.COMMA, ",", "", 1, 0, 0),
+                                Token(
+                                    secondAddressMode.third,
+                                    secondAddressMode.second,
+                                    "",
+                                    1,
+                                    0,
+                                    0,
+                                ),
+                                Token(
+                                    TokenType.NUMBER,
+                                    secondAddress.toString(),
+                                    secondAddress,
+                                    1,
+                                    0,
+                                    0,
+                                ),
+                                Token(TokenType.EOF, "", "", 1, 0, 0),
+                            )
+
+                        val expectedInstance =
+                            instruction.second.constructors
+                                .first()
+                                .call(
+                                    firstAddress,
+                                    secondAddress,
+                                    firstAddressMode.first,
+                                    secondAddressMode.first,
+                                    instruction.third,
+                                )
+
+                        inputWithExpected.add(Pair(tokens, expectedInstance))
+                    }
+                }
+            }
+
+            return inputWithExpected.map { Arguments.of(it.first, it.second) }
+        }
+
+        // MOV, SEQ, SNE, CMP, ADD, SUB, MUL, DIV, MOD, SLT, LDP, STP
+        private fun `provide A Mode immediate, B whatever then AB`(): List<Arguments> {
```

Comment:
The "then AB" at the end is a bit confusing to read. Further down you changed it to "then Modifier is ..." in some function names, I would prefer that for clarity.